dashboard/test/ui/features/step_definitions/pd.rb

```diff
@@ -126,43 +126,25 @@
   }
 end
 
-Given(/^I am a facilitator with completed courses$/) do
-  random_name = "TestFacilitator" + SecureRandom.hex[0..9]
-  steps %Q{
-    And I create a teacher named "#{random_name}"
-    And I make the teacher named "#{random_name}" a facilitator for course "CS Fundamentals"
-    And I create a workshop for course "CS Fundamentals" facilitated by "#{random_name}" with 5 people and end it
-  }
-end
-
-Given(/^I am an organizer with completed courses$/) do
-  random_name = "TestOrganizer" + SecureRandom.hex[0..9]
-  steps %Q{
-    And I create a teacher named "#{random_name}"
-    And I make the teacher named "#{random_name}" a workshop organizer
-    And I create a workshop for course "CS Fundamentals" organized by "#{random_name}" with 5 people and end it
-  }
-end
-
 And(/^I make the teacher named "([^"]*)" a facilitator for course "([^"]*)"$/) do |name, course|
   require_rails_env
 
-  user = User.find_by(email: @users[name][:email])
+  user = find_test_user_by_name(name)
   user.permission = UserPermission::FACILITATOR
   Pd::CourseFacilitator.create(facilitator_id: user.id, course: course)
 end
 
 And(/^I make the teacher named "([^"]*)" a workshop organizer$/) do |name|
   require_rails_env
 
-  user = User.find_by(email: @users[name][:email])
+  user = find_test_user_by_name(name)
   user.permission = UserPermission::WORKSHOP_ORGANIZER
 end
 
 And(/^I make the teacher named "([^"]*)" a workshop admin$/) do |name|
   require_rails_env
 
-  user = User.find_by(email: @users[name][:email])
+  user = find_test_user_by_name(name)
```

Comment:
Most of these replacements are simply a refactor, since they were already looking the test user up by email.